src/subquery.js

```diff
@@ -41,9 +41,16 @@ export default _.curry(
     // Set validation dependency to block search, but uses onMarkForUpdate instead
     // so the targetNode can be marked for update before sourceNode resolves.
     // Validate blocks markedForUpdate but onMarkForUpdate does not.
-    targetNode.onMarkForUpdate = () => sourceNode.updatingPromise
+
+    // targetNode.onMarkForUpdate = () => sourceNode.updatingPromise
+
     // This version would not mark targetNode for update until sourceNode is done:
-    // targetNode.validate = () => sourceNode.updatingPromise.then(() => true)
+    // Using setTimeout because updatingPromise is initialized after validation stage
+    targetNode.validate = async () => {
+      await new Promise(resolve => setTimeout(resolve))
```

Comment:
It feels like there’s a better solution for this - it’s only one line and has a comment, but I wonder if there’s a better solution here 🤔 

Also, what happened to the self reactor dispatch?

Reply:
It makes the whole UI behave like there is `debounce: 0`, every click on facet options dispatches a new search, because validate is getting run every time something is happening